python/lsst/ts/hvac/xml/hvac_mqtt_to_SAL_XML.py

```diff
@@ -387,34 +461,23 @@ def _generic_collect_topics_and_items(row, topics, items):
             item_found = False
             for hvac_item in items:
                 if hvac_item.value == item:
-                    idl_type = (
-                        "boolean" if row["range"] == "TRUE (O) FALSE" else "float"
-                    )
-                    if row["unit"] == "-":
-                        unit = "unitless"
-                    elif row["unit"] == "°C":
-                        unit = "deg_C"
-                    elif row["unit"] == "bar":
-                        unit = "bar"
-                    elif row["unit"] == "%":
-                        unit = "%"
-                    else:
-                        raise ValueError(f"Unknown unit {row['unit']}")
                     topics[hvac_topic.name][hvac_item.name] = {
                         "idl_type": idl_type,
                         "unit": unit,
                     }
                     hvac_topics[topic_and_item] = {
                         "idl_type": idl_type,
-                        "unit": unit.strip(),
-                        "topic_type": row["rw"].strip(),
-                        "limits": _parse_limits(row["limits"].strip()),
+                        "unit": unit,
+                        "topic_type": topic_type,
+                        "limits": limits,
                     }
 
                     item_found = True
                     break
             if not item_found:
-                raise ValueError(f"TelemetryItem '{item}' for {topic} not found.")
+                raise ValueError(
+                    f"TelemetryItem '{item}' for {topic} not found in {topic_and_item}"
+                )
```

Comment:
A good suggestion but I wrote the code like this on purpose to be able to get ALL the differences and not only the first. That way I can simply look at the error messages after running the test, collect those and send them to the provider of the CSV and JSON files so they can be corrected. In any case, this unit test is temporary and will be removed as soon as the JSON file has been verified against the actual HVAC server.

Reply:
I believe my suggested change makes no difference at all to the behavior of the code. It just eliminates a variable.

Not that I am insisting you make the change -- far from it. I just wanted to be clear on the effect it would have.

Reply:
Thanks for the explanation. I get it now and actually adopted the change.